lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -1303,6 +1307,45 @@ getFeePerByteFromWalletPParams pp =
   where
     LinearFee LinearFunction{slope} = getFeePolicy $ txParameters pp
 
+-- | Like the 'TxConstraints' field 'txRewardWithdrawalCost', but with added
+-- support for shared wallets via the 'CA.ScriptTemplate' argument.
+--
+-- We may or may not want to support shared wallets in the full txConstraints.
+_txRewardWithdrawalCost
+    :: Write.FeePerByte
+    -> Either CA.ScriptTemplate TxWitnessTag
+    -> Coin
+    -> Coin
+_txRewardWithdrawalCost feePerByte witType =
+    toWallet
+    . Write.feeOfBytes feePerByte
+    . unTxSize
+    . _txRewardWithdrawalSize witType
+
+-- | Like the 'TxConstraints' field 'txRewardWithdrawalSize', but with added
+-- support for shared wallets via the 'CA.ScriptTemplate' argument.
+--
+-- We may or may not want to support shared wallets in the full txConstraints.
+_txRewardWithdrawalSize
+    :: Either CA.ScriptTemplate TxWitnessTag
+    -> Coin
+    -> TxSize
+_txRewardWithdrawalSize _ (Coin 0) = TxSize 0
+_txRewardWithdrawalSize witType _ = TxSize
+        $ fromMaybe (error "txRewardWithdrawalSize: negative size")
```

Comment:
```suggestion
        $ fromMaybe (error "txRewardWithdrawalSize: negative TxSize")
```

Reply:
Wouldn't that be misleadingly precise, and make it sound like `TxSize` was passed in as an argument and unexpectedly was negative? Whereas `TxSize` is a `Natural` and non-negative by construction.